gp-includes/things/note.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Things: GP_Note class
+ *
+ * @package GlotPress
+ * @subpackage Things
+ * @since 3.0.0
+ */
+
+/**
+ * Core class used to implement the notes system.
+ *
+ * @since 3.0.0
+ */
+class GP_Note extends GP_Thing {
+
+	/**
+	 * Name of the database table.
+	 *
+	 * @var string $table_basename
+	 */
+	public $table_basename = 'gp_notes';
+
+	/**
+	 * List of field names for a translation.
+	 *
+	 * @var array $field_names
+	 */
+	public $field_names = array(
+		'id',
+		'original_id',
+		'translation_set_id',
+		'note',
+		'user_id',
+		'date_added',
+		'date_modified',
+	);
+
+	/**
+	 * List of field names which have an integer value.
+	 *
+	 * @var array $int_fields
+	 */
+	public $int_fields = array(
+		'id',
+		'original_id',
+		'translation_set_id',
+		'user_id',
+	);
+
+	/**
+	 * List of field names which cannot be updated.
+	 *
+	 * @var array $non_updatable_attributes
+	 */
+	public $non_updatable_attributes = array( 'id' );
+
+	/**
+	 * SQL string for order by date
+	 *
+	 * @var array $default_order
+	 */
+	public $default_order = 'ORDER BY date_added DESC';
+
+	/**
+	 * Save the note
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param string $args    Parameters that are not used.
+	 *
+	 * @return object The output of the query.
+	 */
+	public function save( $args = null ) {
+		global $wpdb;
+		$original_id    = gp_post( 'original_id' );
+		$translation_id = gp_post( 'translation_id' );
+		$note           = trim( gp_post( 'note' ) );
+		$translation    = new GP_Translation( array( 'id' => $translation_id ) );
+		if ( ! GP::$permission->current_user_can( 'approve', 'translation', $translation_id, array( 'translation' => $translation ) ) ) {
```

Comment:
By default the "thing" classes should not include cap checks. This should be done before calling the save() method.

Reply:
Do you have any suggestion how to do that?
This is a class extended so I am not sure where I have to put that code.

Reply:
You do it before you call the method, for example if your "saving" a note via a put route, then in the `new_post()` route method you'd do the permissions check before calling the save method on the note thing class.